source-dynamodb/discovery.go

```diff
@@ -264,29 +284,11 @@ func discoverTable(ctx context.Context, c *client, table string) (discoveredTabl
 		return discoveredTable{}, false, nil
 	}
 
-	// Make sure the stream is active.
-	streamDescribe, err := c.stream.DescribeStream(ctx, &dynamodbstreams.DescribeStreamInput{
-		StreamArn: tableDescribe.Table.LatestStreamArn,
-		Limit:     aws.Int32(1), // Don't need to list all of the shards right now.
-	})
-	if err != nil {
-		return discoveredTable{}, false, fmt.Errorf("describe stream: %w", err)
-	}
-
-	if streamDescribe.StreamDescription.StreamStatus != streamTypes.StreamStatusEnabled {
-		log.WithFields(log.Fields{
-			"table":     tableDescribe.Table.TableName,
-			"streamArn": tableDescribe.Table.LatestStreamArn,
-			"status":    streamDescribe.StreamDescription.StreamStatus,
-		}).Warn("streaming enabled for table but stream was not active; table will not be captured")
-		return discoveredTable{}, false, nil
-	}
-
-	if streamDescribe.StreamDescription.StreamViewType != streamTypes.StreamViewTypeNewAndOldImages {
+	if tableDescribe.Table.StreamSpecification.StreamViewType != types.StreamViewTypeNewAndOldImages {
```

Comment:
Is it also possible that a table's `StreamSpecification` sets `StreamViewType` to something other than `StreamViewTypeNewAndOldImages`, but the last stream of the table is made with `StreamViewTypeNewAndOldImages`?

i.e.

```
streamDescribe.StreamViewType == StreamViewTypeNewAndOldImages
tableDescribe.StreamSpec.StreamViewType != StreamViewTypeNewAndOldImages
```

Reply:
I don't believe so - per the [API docs](https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_TableDescription.html), the `StreamSpecification` for the `TableDescription` is for the _current_ stream on the table, and you can only have a single stream for a table. And by all accounts even the `Table.StreamSpecification.StreamEnabled` should be a direct replacement of `streamDescribe.StreamDescription.StreamStatus != streamTypes.StreamStatusEnabled`.